src/Tools/dotnet-monitor/Egress/Configuration/EgressPropertiesProvider.cs

```diff
@@ -19,6 +21,12 @@ public EgressPropertiesProvider(IEgressPropertiesConfigurationProvider provider)
             _provider = provider;
         }
 
+        /// <inheritdoc/>
+        public IEnumerable<string> GetKeys()
```

Comment:
Maybe just return a `Dictionary<string, string>` that's already populated all of the values. The current usage in your changes seems that it is getting the keys and then one-by-one getting values via `TryGetPropertyValue`. I think evolution of egress will eliminate the need for `TryGetPropertyValue` because dotnet monitor itself will not need to pull out individual values.